### ISSUE TITLE
Complex expression with "ignore-arrow-function-shorthand" option of "no-void-expression"

### ISSUE THREAD

**Issue by simark** [palantir/tslint#4297]
### Bug Report

- __TSLint version__: 5.11.0
- __TypeScript version__: 3.1.6
- __Running TSLint via__: CLI

#### TypeScript code being linted

```ts
const a = [1,2,3];
a.forEach(v => v && console.log(v));
```

with `tslint.json` configuration:

```json
{
    "defaultSeverity": "error",
    "rules": {
      "no-void-expression": [true, "ignore-arrow-function-shorthand"]
    }
}
```

#### Actual behavior

I get this error:

    ERROR: /tmp/paul/index.ts[2, 21]: Expression has type `void`. Put it on its own line as a statement.

#### Expected behavior

I believe that this should be accepted, due to `"ignore-arrow-function-shorthand"`.
